src/storage-preview/azext_storage_preview/_validators.py

```diff
@@ -539,3 +568,51 @@ def validate_directory_name(cmd, namespace):
         raise ValueError('usage error: The specified --directory-path already exists in current container. '
                          'Please change to a valid blob directory name. If you want to rename a directory, '
                          'please use `az storage blob directory move` command.')
+
+
+def add_progress_callback(cmd, namespace):
```

Comment:
consider using add_upload_progress_callback() because we will need a download progress callback, too.

Reply:
May I ask the upload and download commands in track 2 can not reuse the `add_progress_callback`?
I see that they reuse the `add_progress_callback` in track 1.

Reply:
yes. In track 1 they could use one callback, but in track2 the key word in response are different.

Reply:
OK